ws-client/src/client.rs

```diff
@@ -35,48 +35,68 @@ use jsonrpsee_types::{
 	error::Error,
 	jsonrpc::{self, JsonValue, SubscriptionId},
 };
-use std::convert::TryInto;
 use std::time::Duration;
+use std::{borrow::Cow, convert::TryInto};
 use std::{io, marker::PhantomData};
 
 /// Client that can be cloned.
 ///
-/// > **Note**: This struct is designed to be easy to use, but it works by maintaining a background
-/// >           task running in parallel. If this is not desirable, you are encouraged to use the
-/// >           [`RawClient`] struct instead.
+/// > **Note**: This struct is designed to be easy to use, but it works by maintaining a background task running in parallel.
 #[derive(Clone)]
 pub struct WsClient {
 	/// Channel to send requests to the background task.
 	to_back: mpsc::Sender<FrontToBack>,
-	/// Config.
-	config: WsConfig,
+	/// Request timeout
+	request_timeout: Option<Duration>,
 }
 
-#[derive(Copy, Clone, Debug)]
 /// Configuration.
-pub struct WsConfig {
-	/// Backend channel for serving requests and notifications.
-	pub request_channel_capacity: usize,
-	/// Backend channel for each unique subscription.
-	pub subscription_channel_capacity: usize,
+#[derive(Clone, Debug)]
+pub struct WsConfig<'a> {
+	/// URL to connect to.
+	pub url: &'a str,
 	/// Max request body size
 	pub max_request_body_size: usize,
 	/// Request timeout
 	pub request_timeout: Option<Duration>,
+	/// Connection timeout
+	pub connection_timeout: Duration,
+	/// `Origin` header to pass during the HTTP handshake. If `None`, no
+	/// `Origin` header is passed.
+	pub origin: Option<Cow<'a, str>>,
+	/// Url to send during the HTTP handshake.
+	pub handshake_url: Cow<'a, str>,
+	/// Max concurrent request capacity.
+	///
+	/// **Note**: The actual capacity is `num_senders + max_concurrent_requests_capacity`
+	/// because it is passed to [`futures::mpsc::channel`](futures::channel::mpsc::channel)
+	/// and the capacity may increase because the sender is cloned when new
+	/// requests, notifications and subscriptions are created.
+	pub max_concurrent_requests_capacity: usize,
+	/// Max concurrent capacity for each subscription which might be dropped if the capacity is exceeded.
```

Comment:
Something's odd with this phrase after "which". Do you mean to say "Max concurrent capacity for each subscription; when exceeded messages will be dropped."?